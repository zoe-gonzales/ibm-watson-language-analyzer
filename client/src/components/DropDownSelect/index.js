import React from 'react';

const DropDown = ({
    name,
    options,
    onChange
}) => {
    return (
        <div>
            <select id={name} onChange={onChange}>
                {options.map(opt => <option value={opt.name} key={opt.id}>{opt.name}</option>)}
            </select>
            <label for={name}>Select...</label>
        </div>
        
    )
}

export default DropDown;