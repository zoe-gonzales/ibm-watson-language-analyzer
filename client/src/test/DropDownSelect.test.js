import React from 'react';
import DropDownSelect from '../components/DropDownSelect';
import renderer from 'react-test-renderer';

it('renders DropDownSelect component correctly', () => {
    const onChange = e => console.log('something has changed and the event is this:' + e);
    const name = 'opts'
    const options = [
        {
            'name': 'Option 1',
            'id': 2
        },
        {
            'name': 'Option 2',
            'id': 4
        },
        {
            'name': 'Option 3',
            'id': 6
        },
        {
            'name': 'Option 4',
            'id': 8
        } 
    ];
    const component = renderer
    .create(<DropDownSelect options={options} name={name} onChange={onChange} />)
    .toJSON();
    expect(component).toMatchSnapshot();
});