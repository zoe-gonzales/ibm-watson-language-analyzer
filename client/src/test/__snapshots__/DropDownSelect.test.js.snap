// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DropDownSelect component correctly 1`] = `
<div>
  <select
    id="opts"
    onChange={[Function]}
  >
    <option
      value="Option 1"
    >
      Option 1
    </option>
    <option
      value="Option 2"
    >
      Option 2
    </option>
    <option
      value="Option 3"
    >
      Option 3
    </option>
    <option
      value="Option 4"
    >
      Option 4
    </option>
  </select>
  <label
    for="opts"
  >
    Select...
  </label>
</div>
`;
